mcstatus/__main__.py

```diff
@@ -1,75 +1,107 @@
 from __future__ import annotations
 
+import sys
 import argparse
 import socket
+from typing import TYPE_CHECKING
 from json import dumps as json_dumps
 
-from mcstatus import JavaServer
+from mcstatus import JavaServer, BedrockServer
+from mcstatus.responses import JavaStatusResponse
 
+if TYPE_CHECKING:
+    SupportedServers = JavaServer | BedrockServer
 
-def ping(server: JavaServer) -> None:
-    print(f"{server.ping()}ms")
 
+def ping(server: SupportedServers) -> int:
+    # this method supports both Java and Bedrock.
+    # only Java supports the `ping` packet, and even then not always:
+    # https://github.com/py-mine/mcstatus/issues/850
+    print(f"{server.status().latency}")
+    return 0
 
-def status(server: JavaServer) -> None:
+
+def status(server: SupportedServers) -> int:
     response = server.status()
-    if response.players.sample is not None:
-        player_sample = str([f"{player.name} ({player.id})" for player in response.players.sample])
+
+    java_res = response if isinstance(response, JavaStatusResponse) else None
+
+    if not java_res:
+        player_sample = ""
+    elif java_res.players.sample is not None:
+        player_sample = str([f"{player.name} ({player.id})" for player in java_res.players.sample])
     else:
         player_sample = "No players online"
 
-    print(f"version: v{response.version.name} (protocol {response.version.protocol})")
-    print(f'motd: "{response.motd}"')
-    print(f"players: {response.players.online}/{response.players.max} {player_sample}")
+    if player_sample:
+        player_sample = " " + player_sample
 
+    print(f"version: {server.kind()} {response.version.name} (protocol {response.version.protocol})")
+    print(f"motd: {response.motd.to_ansi()}")
```

Comment:
Since most Minecraft servers provide two lines of MOTD, this could use

```suggestion
    print(f"motd:\n{response.motd.to_ansi()}")
```
![image](https://github.com/user-attachments/assets/2b8061db-c6b0-4e1e-958b-bb789bcbf6b7)